vulture/core.py

```diff
@@ -91,14 +91,22 @@ def _ignore_variable(filename, varname):
         _is_special_name(varname))
 
 
-class Item(str):
-    def __new__(cls, name, typ, filename, lineno, size=1):
-        item = str.__new__(cls, name)
-        item.typ = typ
-        item.filename = filename
-        item.lineno = lineno
-        item.size = size
-        return item
+class Item(object):
+    def __init__(self, name, typ, filename, lineno, size=1):
+        self.name = name
+        self.typ = typ
+        self.filename = filename
+        self.lineno = lineno
+        self.size = size
+
+    def __str__(self):
+        return self.name
+
+    def __eq__(self, other):
+        return self.name == other
```

Comment:
You'll probably have to change the set difference operation in _get_unused_items() since some lists contain Items and others contain strings. I just pushed a commit which unifies things: All defined_* lists always contained Item objects, all remaining lists ("usages") now contain plain strings. What we probably need is

```
def _get_unused_items(defined_items, used_names):
    used_names = set(used_names)
    unused_items = [item for item in set(defined_items) if item.name not in used_names]
    unused_items.sort(key=lambda item: item.name.lower())
    return unused_items
```